src/components/ListPage/ListComponent.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import { AxiosError } from 'axios';
+import { ListPage } from './ListPage';
+import { SortField } from '../../types/SortFilters';
+import { Pagination } from '../../types/Pagination';
+
+export namespace ListComponent {
+  type ListComponentProps<R> = {
```

Comment:
You could have it defined as an interface, and props from child classes (like `AppListComponentProps`) would extend it, so that there's no need to repeat all the fields.

Some fields may be specific to the child classes, like here `rateInterval` isn't used by `IstioConfigListComponent` so in such situation I think it shouldn't appear here, and leave the related code to the child classes even if it's sometimes (but not always) redundant.